internal/v1/handler_recommend_package_test.go

```diff
@@ -0,0 +1,57 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/osbuild/image-builder/internal/recommendations"
+	"github.com/osbuild/image-builder/internal/tutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRecommendPackage_Success(t *testing.T) {
+	apiSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if "Bearer" == r.Header.Get("Authorization") {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		require.Equal(t, "Bearer accesstoken", r.Header.Get("Authorization"))
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		message := []string{"vim", "python"}
+		result := recommendations.Response{Message: &message}
+		err := json.NewEncoder(w).Encode(result)
+		require.NoError(t, err)
+	}))
+	defer apiSrv.Close()
+
+	srv, tokenSrv := startServer(t, "", "", apiSrv.URL, &ServerConfig{})
+	defer func() {
+		err := srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	}()
+	defer tokenSrv.Close()
+	payload := recommendations.UploadPackageRequest{
+		Packages: &[]string{
+			"some",
+			"packages",
+		},
+		RecommendedPackages: func() *int32 {
+			recommendedPackages := int32(3)
+			return &recommendedPackages
+		}(),
+	}
+
+	//payload := RecommendationsResponse{
+	//	Message: &[]string{"vim", "python"},
+	//}
+	respStatusCode, body := tutils.PostResponseBody(t, "http://localhost:8086/api/image-builder/v1/experimental/recommendations", payload)
+	require.Equal(t, http.StatusCreated, respStatusCode)
```

Comment:
This endpoint should have never been POST, this should be GET and return just HTTP OK 200. This needs also to be fixed in the service first.